packages/go/stbernard/graph/transform.go

```diff
@@ -0,0 +1,346 @@
+func (s *command) Usage() string {
+	return Usage
+}
+
+// Name of command
+func (s *command) Name() string {
+	return Name
+}
+
+// Parse command flags
+func (s *command) Parse(cmdIndex int) error {
+	cmd := flag.NewFlagSet(Name, flag.ExitOnError)
+	workspacePaths, err := workspace.FindPaths(s.env)
+	if err != nil {
+		return fmt.Errorf("failed to find workspace paths: %w", err)
+	}
+	s.root = workspacePaths.Root
+
+	path := cmd.String("path", workspacePaths.Root, "destination path for arrows.json file, default is root")
+
+	cmd.Usage = func() {
+		w := flag.CommandLine.Output()
+		fmt.Fprintf(w, "%s\n\nUsage: %s %s [OPTIONS]\n\nOptions:\n", Usage, filepath.Base(os.Args[0]), Name)
+		cmd.PrintDefaults()
+	}
+
+	if err := cmd.Parse(os.Args[cmdIndex+1:]); err != nil {
+		cmd.Usage()
+		return fmt.Errorf("parsing %s command: %w", Name, err)
+	}
+
+	if *path != "" {
+		s.path = *path
+	}
+
+	return nil
+}
+
+// Run generate command
+func (s *command) Run() error {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	var (
+		outPath = filepath.Join(s.path, "arrows.json")
+	)
+
+	if database, err := s.initializeDatabase(ctx); err != nil {
+		return fmt.Errorf("error connecting to database: %w", err)
+	} else if ingestFilePaths, err := s.getIngestFilePaths(); err != nil {
+		return fmt.Errorf("error getting ingest file paths from test directory %v", err)
+	} else if err = ingestData(ctx, ingestFilePaths, database); err != nil {
+		return fmt.Errorf("error ingesting data %v", err)
+	} else if nodes, edges, err := getNodesAndEdges(ctx, database); err != nil {
+		return fmt.Errorf("error retrieving nodes and edges from database %w", err)
+	} else if arrows, err := transformToArrows(nodes, edges); err != nil {
+		return fmt.Errorf("error transforming nodes and edges to arrows format %w", err)
+	} else if jsonBytes, err := json.MarshalIndent(arrows, "", "  "); err != nil {
+		return fmt.Errorf("error occurred while marshalling arrows into bytes %w", err)
+	} else if err := os.MkdirAll(filepath.Dir(outPath), 0o755); err != nil {
+		return fmt.Errorf("creating output directory: %w", err)
+	} else {
+		return os.WriteFile(outPath, jsonBytes, 0o644)
+	}
+}
+
+type Position struct {
+	X int `json:"x"`
+	Y int `json:"y"`
+}
+
+type Node struct {
+	ID         string            `json:"id"`
+	Position   Position          `json:"position"`
+	Caption    string            `json:"caption"` // name of node -- if not, use object_id
+	Label      []string          `json:"label"`   // kinds
+	Properties map[string]string `json:"properties"`
+}
+
+type Relationship struct {
+	ID         string            `json:"id"`
+	Label      string            `json:"label"` // kind
+	From       string            `json:"fromId"`
+	To         string            `json:"toId"`
+	Properties map[string]string `json:"properties"`
+}
+
+type Arrows struct {
+	Nodes         []Node         `json:"nodes"`
+	Relationships []Relationship `json:"relationships"`
+}
```

Comment:
Since Arrows isn't an appropriate tool for visualizing these graphs, and we ultimately need to import them, we should restructure to make compatible with our generic graph tools